src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDiffieHellman/ECDiffieHellmanTests.ImportExport.cs

```diff
@@ -416,6 +417,48 @@ public static void ImportFromPrivateOnlyKey()
             }
         }
 
+        [Theory]
+        [MemberData(nameof(NamedCurves))]
+        public static void OidPresentOnCurveMiscased(ECCurve curve)
+        {
+            ECCurve miscasedCurve = ECCurve.CreateFromFriendlyName(InvertStringCase(curve.Oid.FriendlyName));
+            Assert.NotEqual(miscasedCurve.Oid.FriendlyName, curve.Oid.FriendlyName);
+            Assert.Equal(miscasedCurve.Oid.FriendlyName, curve.Oid.FriendlyName, ignoreCase: true);
+
+            using (ECDiffieHellman ecdh = ECDiffieHellmanFactory.Create())
+            {
+                ecdh.GenerateKey(miscasedCurve);
+                ECParameters exportedParameters = ecdh.ExportParameters(false);
+                Assert.Equal(curve.Oid.Value, exportedParameters.Curve.Oid.Value);
+
+                exportedParameters.Curve = miscasedCurve;
+                ecdh.ImportParameters(exportedParameters);
```

Comment:
Is this just what I often call `// Assert.NoThrow`, or did you mean to do more after it?  (The only thing I can really think of for after is showing that exportedParameters2's curve is cased like... whichever one it is :smile:)

Reply:
Your first guess is right. Will clarify with a comment. (Will get that checked in with on the outcome of the other discussion)